frontend/src/metabase/components/Popover/TippyPopover.tsx

```diff
@@ -107,13 +110,15 @@ function TippyPopover({
 
   const handleShow = useCallback(
     (instance: TippyInstance) => {
-      setupCloseHandler(instance.popper, () => instance.hide());
+      setupCloseHandler(instance.popper, () =>
+        isControlled ? onClose?.() : instance.hide(),
```

Comment:
When a parent component is controlling a popover via its `visible` prop, we can't use tippy instance methods like `hide` or `show` because `instance.hide()` and setting `visible` from `true` to `false` both trigger tippy's `onHide` function, so `onHide` would trigger twice.

I'd hate to have to disable `TippyPopover`'s use of the hook `useSequencedContentCloseHandler` completely when a popover is controlled via `visible` (that functionality would need to be reimplemented in the parent component), so I'm adding an additional function prop, `onClose`, that needs to be added if from a parent component you want both to control the popover's visibility and to have it close itself when the user presses the escape key (presumably you do).

You can see in `TippyPopoverWithTrigger` how it is used.